src/smac_planner_hybrid.cpp

```diff
@@ -81,6 +82,28 @@ void SmacPlannerHybrid::initialize(
   dsrv_->setCallback(boost::bind(&SmacPlannerHybrid::reconfigureCB, this, _1, _2));
 }
 
+bool SmacPlannerHybrid::arePosesSameDiscreteState(
+  const geometry_msgs::Pose& pose1,
+  const geometry_msgs::Pose& pose2,
+  costmap_2d::Costmap2D* costmap) const
+{
+  // Convert world coordinates to map coordinates
+  float pose1_mx, pose1_my, pose2_mx, pose2_my;
+  costmap->worldToMapContinuous(pose1.position.x, pose1.position.y, pose1_mx, pose1_my);
+  costmap->worldToMapContinuous(pose2.position.x, pose2.position.y, pose2_mx, pose2_my);
+
+  // Get orientation bins (using NodeHybrid's method)
+  double pose1_orientation = tf2::getYaw(pose1.orientation);
+  double pose2_orientation = tf2::getYaw(pose2.orientation);
+  unsigned int pose1_bin_id = NodeHybrid::motion_table.getClosestAngularBin(pose1_orientation);
```

Comment:
S: new variable not needed
```suggestion
  unsigned int pose1_bin_id = NodeHybrid::motion_table.getClosestAngularBin(tf2::getYaw(pose1.orientation));
```